IndicoToolkit/AutoReview/AutoReviewFunctions.cs

```diff
@@ -0,0 +1,225 @@
+using System.Collections.Generic;
+
+using IndicoToolkit.Types;
+
+namespace IndicoToolkit.AutoReview
+{
+    /// <summary>
+    /// Class that hosts suite of auto review functions
+    /// </summary>
+    public class AutoReviewFunctions
+    {
+        public AutoReviewFunctions() { }
+
+        /// <summary>
+        /// Determine whether prediction label matches a label in labels or labels list is empty
+        /// </summary>
+        /// <param name="prediction">prediction to check label</param>
+        /// <param name="labels">List of labels</param>
+        /// <returns>
+        /// true labels list is empty or prediction label matches a label in the list
+        /// </returns>
+        public bool emptyOrMatchingLabel(Prediction prediction, List<string> labels)
+        {
+            if (labels == null || labels.Count == 0 || labels.Contains(prediction.getLabel()))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Rejects predictions below a given confidence threshold
+        /// </summary>
+        /// <param name="predictions">List of predictions</param>
+        /// <param name="labels">List of labels</param>
+        /// <param name="confThreshold">Threshold to reject predictions</param>
+        /// <returns>
+        /// all predictions 
+        /// </returns>
+        public List<Prediction> rejectByConfidence(List<Prediction> predictions, List<string> labels = default, float confThreshold = .5f)
+        {
+            foreach (Prediction prediction in predictions)
+            {
+                if (prediction.getValue("rejected") == null)
+                {
+                    if (!emptyOrMatchingLabel(prediction, labels))
+                    {
+                        continue;
+                    }
+                    string predLabel = prediction.getLabel();
+                    if (prediction.getValue($"confidence.{predLabel}") < confThreshold)
+                    {
+                        prediction.setValue("rejected", true);
+                        prediction.removeKey("accepted");
+                    }
+                }
+            }
+            return predictions;
+        }
+
+        /// <summary>
+        /// Removes predictions below a given confidence threshold
+        /// </summary>
+        /// <param name="predictions">List of predictions</param>
+        /// <param name="labels">List of labels</param>
+        /// <param name="confThreshold">Threshold to remove predictions</param>
+        /// <returns>
+        /// all predictions above confThreshold. 
+        /// </returns>
+        public List<Prediction> removeByConfidence(List<Prediction> predictions, List<string> labels = default, float confThreshold = .5f)
```

Comment:
what's the difference between remove/reject by confidence practically speaking?

Reply:
Remove would "remove clutter" from the output,  Reject just notifies user that prediction is invalid. I thought that there could be a case where a user would want to just not see certain predictions if they know the predictions will not be relevant at all (i.e. clean up output). but I can see how it could be better to just provide reject and assume if a user wants to clean up predictions they can do it themselves

Reply:
maybe we should have a call to discuss